src/App.tsx

```diff
@@ -1,39 +1,58 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import Clock from './components/Clock';
+
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.oncontextmenu = (event) => {
+      event.preventDefault();
+    };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('mouseup', this.handleClick);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  handleClick = (event: MouseEvent) => {
+    if (event.button === 0) {
+      this.setState({ hasClock: true });
+    }
 
-        {' time is '}
+    if (event.button === 2) {
+      this.setState({ hasClock: false });
+    }
+  };
```

Comment:
i am not sure about this solution. I think it's OK)
In readme.md recomended to use (`contextmenu` event)